components/domains/autorenewal.tsx

```diff
@@ -0,0 +1,318 @@
+    // salt must not be empty to preserve privacy
+    if (!salts) return;
+
+    const computeHashes = async () => {
+      const metaDataHashes = await Promise.all(
+        salts.map((salt) =>
+          computeMetadataHash(
+            email,
+            isSwissResident ? "switzerland" : "none",
+            salt
+          )
+        )
+      );
+      setMetadataHashes(metaDataHashes);
+    };
+
+    computeHashes();
+  }, [email, salts, renewalBox, isSwissResident]);
+
+  useEffect(() => {
+    if (!selectedDomains) return;
+    setPrice(
+      getPriceFromDomains(
+        selectedDomainsToArray(selectedDomains),
+        duration
+      ).toString()
+    );
+  }, [selectedDomains, duration]);
+
+  function changeEmail(value: string): void {
+    setEmail(value);
+    setEmailError(isValidEmail(value) ? false : true);
+  }
+
+  useEffect(() => {
+    if (isSwissResident) {
+      setSalesTaxRate(swissVatRate);
+      setSalesTaxAmount(applyRateToBigInt(price, swissVatRate));
+    } else {
+      setSalesTaxRate(0);
+      setSalesTaxAmount("");
+    }
+  }, [isSwissResident, price]);
+
+  useEffect(() => {
+    if (selectedDomains && metadataHashes) {
+      const calls = [];
+
+      if (
+        erc20AllowanceError ||
+        (erc20AllowanceData &&
+          erc20AllowanceData["remaining"].low !== UINT_128_MAX &&
+          erc20AllowanceData["remaining"].high !== UINT_128_MAX)
+      ) {
+        calls.push(autoRenewalCalls.approve());
+      }
+
+      selectedDomainsToArray(selectedDomains).map((domain, index) => {
+        const encodedDomain = utils
+          .encodeDomain(domain)
+          .map((element) => element.toString())[0];
+        const price = getPriceFromDomain(1, domain);
+        const allowance: string = salesTaxRate
+          ? (
+              BigInt(price) + BigInt(applyRateToBigInt(price, salesTaxRate))
+            ).toString()
+          : price.toString();
+        calls.push(
+          autoRenewalCalls.enableRenewal(
+            encodedDomain,
+            allowance,
+            "0x" + metadataHashes?.[index]
+          )
+        );
+      });
+      setCallData(calls);
+    }
+  }, [
+    selectedDomains,
+    price,
+    salesTaxAmount,
+    erc20AllowanceData,
+    metadataHashes,
+    salesTaxRate,
+    erc20AllowanceError,
+  ]);
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.card}>
+        <div className={styles.form}>
+          <BackButton onClick={() => router.back()} />
+          <div className="flex flex-col items-start gap-0 self-stretch">
+            <h3 className={styles.domain}>Enable subscription</h3>
+            <p className="py-2 text-left">
+              Enable automatic domain renewal to ensure uninterrupted ownership
+              and benefits. Never worry about expiration dates again.
+            </p>
+          </div>
+          <div className="flex flex-col items-start gap-6 self-stretch">
+            <TextField
```

Comment:
We need to make this based on the fact that the user has or doesn't have metadata already